mc-sema/cfgToLLVM/raiseX86_64.cpp

```diff
@@ -273,102 +277,58 @@ bool addEntryPointDriver(Module *M,
             aliasMCSemaScope(new StoreInst(curArg, arg4, driverBB));
         }
     } else if (getSystemOS(M) == llvm::Triple::Linux) {
-        //#else
-        if(fwd_it != fwd_end) {
-            int   k = x86_64::getRegisterOffset(RDI);
-            Value *rdiFieldGEPV[] = {
-                CONST_V<64>(driverBB, 0),
-                CONST_V<32>(driverBB, k)
+      //#else
+      unsigned fp_reg_num = 0;
+      unsigned notfp_reg_num = 0;
+      Value *args_fp[8];
+      Value *args_notfp[6];
+      int reg_offset_fp[8] = {x86_64::getRegisterOffset(XMM0), x86_64::getRegisterOffset(XMM1), x86_64::getRegisterOffset(XMM2), x86_64::getRegisterOffset(XMM3), x86_64::getRegisterOffset(XMM4), x86_64::getRegisterOffset(XMM5), x86_64::getRegisterOffset(XMM6), x86_64::getRegisterOffset(XMM7)};
+      int reg_offset_notfp[6] = {x86_64::getRegisterOffset(RDI), x86_64::getRegisterOffset(RSI), x86_64::getRegisterOffset(RDX), x86_64::getRegisterOffset(RCX), x86_64::getRegisterOffset(R8), x86_64::getRegisterOffset(R9)};
+
+      while(fwd_it != fwd_end) {
+        Type *T = fwd_it->getType();
+        if(T->isDoubleTy()){
```

Comment:
> What about `isFloatTy()`?

I think that it is important not "whether float/double type" but "whether FloatingPoint/NotFloatingPoint value".
If a "F" word is set in cfg_to_bc "-driver" option, this argument is treated floating point value as double type, according to the codes before my modification.
I just wrote codes to fit this.

> What happens with something like`long double sinl(long double x);`?

 I don't know, sorry :(